### ISSUE TITLE
Andrew to remove ARPACK and "_legacy" eigensolvers.

### ISSUE THREAD

**Issue by MatthiasHeilManchester** [oomph-lib/oomph-lib#81]
@alhazel 

- [x] As I said elsewhere, I needed to retain the "_legacy" postfix for the old versions of the eigensolvers. Please remove calls to them (and then the functions themselves) everywhere once you've upated the relevant demo_drivers to use the new interfaces.

- [x] Also remove temporary boolean
 
         Anasazi::Use_temporary_code_for_andrew_legacy_version

   that's used in the implementation of the "operators" that anasazi needs.

- [x] MH has currently commented out many member functions (and associated private data) in the Anasazi eigensolver. They were neither used nor required from the Eigensolver base class (though they seemed useful!). I have no use case
 for them so can't check them but if they're required (in Anasazi's anticipated replacement for arpack) then they should be reimplemented in such a way that they actually do what they say.

    Note from Andrew to Andrew via Matthias:
```
trilinos_eigen_solver.h, line 772:

Teuchos::ParameterList MyPL;
MyPL.set("Which", "LM");

```
   This will be where the different choices of spectrum (e.g. left of shift, right of shift, etc) will need to be specified. 

- [x] arpack is still around but is longer used in any of the demo drivers. lapack QZ is now the default eigensolver.  Please remove arpack completely once you're happy that none of your private drivers still needed it (not all capabiilties of arpack appear to be accessible in the anasazi reimplementation).

**Comment by puneetmatharu** [comment]
Any updates on this?

**Comment by alhazel** [comment]
Nope, but I should get to it before Christmas. Given how little computing I actually do these days (see delay in getting this done), any drivers with ARPACK involved would probably need to be rewritten anyway.

**Comment by puneetmatharu** [comment]
Okay, sounds good!

**Comment by puneetmatharu** [comment]
Boop.

Did we handle this?

**Comment by MatthiasHeilManchester** [comment]
All done; no references to "_legacy" or "arpack" (with/without captitalisation) in our own files. Apart from one unfortunate reference where @puneetmatharu explains how to use HAVE_ flags. @puneetmatharu : It's in 

./cmake/OomphFindExternalSrcFile.cmake:# As an example, this may be set to HAVE_ARPACK_SOURCES. If the REQUIRED flag

Can you update this and then close this issue, please? (I could make a wild guess as to what to use instead but you'd have to check it anywaym, so...).
